File: 0121-best-time-to-buy-and-sell-stock/0121-best-time-to-buy-and-sell-stock.cpp
class Solution {
public:
    int maxProfit(vector<int>& prices) {
        int min = INT_MAX;
        int profit = 0;
        for(int val : prices){
            if(val < min){
                min = val;
            }
           if ((val - min) > profit){
                profit = val - min;
           }
        }
    return profit;
    }
};

File: 0121-best-time-to-buy-and-sell-stock/0121-best-time-to-buy-and-sell-stock.py
class Solution:
    def maxProfit(self, prices: List[int]) -> int:
        min = prices[0]
        profit = 0
        for i in range(len (prices)):
            if(min > prices[i]):
                min = prices[i]
            if(prices[i] - min ) > profit:
                profit = prices[i] - min
        return profit

            
        